Transport parameters are mandatory. Fixes #2528 (#2560)
quicwg/base-drafts: draft-ietf-quic-tls.md

the version of QUIC defined in {{QUIC-TRANSPORT}} is used.

The quic_transport_parameters extension is carried in the ClientHello and the
EncryptedExtensions messages during the handshake. Endpoints MUST send the

> quic_transport_parameters extension; endpoints which receive ClientHello

Reviewer: ```suggestion
quic_transport_parameters extension; endpoints that receive ClientHello
```
CMOS

> or EncryptedExtensions messages without the transport_parameters extension MUST

Reviewer: ```suggestion
or EncryptedExtensions messages without the quic_transport_parameters extension MUST
```

> terminate the TLS handshake with a fatal missing_extension alert.

Reviewer: ```suggestion
terminate the TLS handshake with a fatal missing_extension alert (an error of 0x16d).
```

the version of QUIC defined in {{QUIC-TRANSPORT}} is used.

The quic_transport_parameters extension is carried in the ClientHello and the
EncryptedExtensions messages during the handshake. Endpoints MUST send the
quic_transport_parameters extension; endpoints that receive ClientHello or
EncryptedExtensions messages without the quic_transport_parameters extension

> MUST terminate the TLS handshake with a fatal missing_extension alert (an error

Reviewer 2: ```suggestion
MUST close the connection with an error of type 0x16d (equivalent to a fatal TLS missing_extension alert, see Section {{tls-errors}}).
```

Reviewer 3: This will result in "Section Section X.Y" in the final output.

Reviewer 3: Also, that comma should be a semicolon; see #2626.

Endpoints MUST NOT send this extension in a TLS connection that does not use
QUIC (such as the use of TLS with TCP defined in {{!TLS13}}).  A fatal
unsupported_extension alert (an error of 0x16e) MUST be sent if this extension

> is received when the transport is not QUIC.

Reviewer 2: This is not just in your text, but while you're here, also replace the text with:
"The connection MUST be closed with an error code of type 0x16e (equivalent to a fatal TLS unsupported_extension alert, see Section {{tls-errors}}) if the transport is not QUIC."

Reviewer 3: Same.

Reviewer 4: I agree with @MikeBishop. Specifying the QUIC error code to send if the extension is received when the transport isn't QUIC doesn't make much sense.
